### DISCUSSION TITLE
record a video on YouTube

### DISCUSSION THREAD

**Discussion by SyntaxEvg** [riok/mapperly#195] [General]
I don’t understand how to make the essence of the DTO through your package, the instructions that are written here are not clear, could you describe the whole example or record a video on **YouTube**
[https://mapperly.riok.app/docs/getting-started/first-mapper](https://mapperly.riok.app/docs/getting-started/first-mapper) - SOS 
example: 
```
[Mapper]
public class Employee{
    public Guid Id { get; }
    public string Name { get; }
    public DateTime Date { get; }
}
```
End: 
```
public class EmployeeDTO {
    public Guid IdEmployee  { get; }
    public string NameEmployee  { get; }
    public DateTime Date { get; }
}
```

**Comment by CommonGuy** [comment]
We currently have no plans to create YouTube video tutorials, as we are not experienced in this. However, we would welcome videos from the community.

Regarding your example, you need to create a new class:
```csharp
[Mapper]
public static partial class EmployeeMapper
{
    // Mapperly will create an implementation for this method
    public static partial EmployeeDTO MapEmployeeToDto(Employee employee);
}
```
which can then be used like this:
```csharp
var employee = new Employee();
var myDto = EmployeeMapper.MapEmployeeToDto(employee);
```

**Reply by SyntaxEvg** [reply]
![image](https://user-images.githubusercontent.com/94603687/207030206-9e27840e-0b88-4c04-9047-76f6576fad2d.png)
Help =(

**Reply by CommonGuy** [reply]
Create a "normal" C# class, which represents your source class (which you want to map to another class). Name the file `Employee.cs`:
```csharp
public class Employee
{
    public Guid Id { get; set; }
    public string Name { get; set; }
    public DateTime Date { get; set; }
}
```

Then, create a "destination class". Name the file `EmployeeDTO.cs`:
```csharp
public class EmployeeDTO
{
    public Guid IdEmployee { get; set; }
    public string NameEmployee { get; set; }
    public DateTime Date { get; set; }
}
```

Next, create the mapper class. Name it `EmployeeMapper.cs`:
```csharp
[Mapper]
public static partial class EmployeeMapper
{
    // Mapperly will create an implementation for this method
    [MapProperty(nameof(Employee.Id), nameof(EmployeeDTO.IdEmployee))] // Since these property names do not match, we need to configure them explicitly
    [MapProperty(nameof(Employee.Name), nameof(EmployeeDTO.NameEmployee))]
    public static partial EmployeeDTO MapEmployeeToDto(Employee employee);
}
```

Now, use the mapper wherever you want:
```csharp
var myEmployee = new Employee
{
    Id = Guid.NewGuid(),
    Name = "my name",
    Date = DateTime.Now,
};
var employeeDto = EmployeeMapper.MapEmployeeToDto(myEmployee);
```

**Comment by latonz** [comment]
Nick Chapsas created a video explaining Mapperly: [YouTube](https://youtu.be/U8gSdQN2jWI?si=Rf9v_SFO_sWnvRsC&t=236)
